### ISSUE TITLE
Feature Request: ability to run external command

### ISSUE THREAD

**Issue by mario-grgic** [vrapper/vrapper#659]
It would be useful to add Vim's silent execute command to vrapper with the ability to pass the current Eclipse workspace, project and file being edited to the external program via variables, or alternatively using Vim's % with filename modifiers %:p, %:h, %:t etc

**Comment by albertdev** [comment]
Is this about `:!` or about `:<range>!`?

The first is requested in #614. The second command is partially supported but lacks the `%` placeholder, that is requested in #599.

**Comment by mario-grgic** [comment]
Yes, essentially `:!external_command` is it. Basically, ability to execute external command passing the current file name with file modifiers, like %:p, %:h etc (or alternatively eclipse workspace location, project, file etc. Eclipse already exposes these as variables) and the result of the execution of the external command should not be inserted in the file being edited. Only the exit status of the command can perhaps be reported in the Eclipse status bar as normal message.

**Comment by keforbes** [comment]
I recently implemented the quick hack of `%` in #599 but now I see this request and it looks like there's more to be included.  This is just a note to myself, I'll need to look at `:help filename-modifiers` to see everything Vim supports.

**Comment by elicoten** [comment]
One thing that would be really useful for my use-case would be to enable support for escaping the `%` placeholder when you don't want it.

For example I'm trying to insert a timestamp using `r! date -d 2017-08-01 +\%s`. It works fine in vim, and I haven't been able to find any other workaround.

**Comment by keforbes** [comment]
@elicoten, I made the requested fix.  Hopefully I'll be able to create an unstable build in a day or two for you to test.

**Comment by keforbes** [comment]
@elicoten, sorry it took so long, I really don't have time to spend on this project anymore.  Anyway, I finally updated the unstable update site with a new build (0.73.20170801) which includes this change.  Please give it a try and let me know if you hit any issues with it.

**Comment by elicoten** [comment]
@keforbes Sorry for the delay in getting back to you. I just tested this fix and it seems to work a treat. From looking at the code it looks like it couldn't handle a mixture of 'real' `%` and escaped `\%` in the same line but for now I don't think I need to do that anyway. (Plus I suppose could always manually copy the file name in there)
